pkg/yurthub/cachemanager/cache_manager.go

```diff
@@ -587,5 +585,24 @@ func (cm *cacheManager) CanCacheFor(req *http.Request) bool {
 		return false
 	}
 
+	cm.Lock()
+	defer cm.Unlock()
+	if info.Verb == "list" && info.Name == "" {
+		key, _ := util.KeyFunc(comp, info.Resource, info.Namespace, info.Name)
+		if _, ok := cm.listReqCollector[key]; !ok {
+			for k := range cm.listReqCollector {
```

Comment:
Can you explain the purpose of having listReqCollector? 

Also, can you give an example that "len(k) > len (key) " and an example for "len(k) < len(key)"?

Reply:
1. because func queryListObject() will get all pods for both requests instead of getting pods by request selector. so cache manager can not support same path list requests that has different selector. for example:
  -  request1: http://{ip:port}/api/v1/default/pods?labelSelector=foo=bar
  - request2: http://{ip:port}/api/v1/default/pods?labelSelector=foo2=bar2

2. because func queryListObject() will get all pods for both requests instead of getting pods by request selector. so cache manager can not support getting same resource list requests that has different path. for example:
  - request1: http://{ip/port}/api/v1/pods?fieldSelector=spec.nodeName=foo
  - request2: http://{ip/port}/api/v1/default/pods?fieldSelector=spec.nodeName=foo

in order to solve the above two problems, we need listReqCollector to keep list request info,  and based on listReqCollector to verify list requests can cache or not.

> Also, can you give an example that "len(k) > len (key) " and an example for "len(k) < len(key)"?
 the second above example.